woocommerce/woocommerce-ios
# Shipping Label: Main customs form validation

Part of #4687
⚠️ This PR change is more than 500 lines long - I blame the generated copiables and my restructuring of SwiftUI stacks! ⚠️

# Description
This PR aims to add validation for the text fields in the customs form: content explanation, restriction comments and ITN. Complete validation for the whole form will be implemented in the next PR to avoid bloating this PR.

# Changes
### Customs form input view model
- Updated initializer to accept new param for destination country.
- Added check to see if destination country requires validation for ITN.
- Added validation checks for content explanation, restriction comments, ITN. The validation for ITN is incomplete - the check for accumulated value of each tariff number will be implemented in the next PR.
- Updated to make sure that content explanation and restriction comments are reset when content type and restriction comments are relatively not `.other`.
- Added unit tests for validations.
### UI changes
- Moved each row into a separate calculated variable to make `body` less bloated.
- Added validation error row for content explanation, restriction comments and  ITN.

# Demo
![form-validation](https://user-images.githubusercontent.com/5533851/129869312-4b44d0ff-98a4-4b6d-93f5-1366599f9f32.gif)

# Testing
1. Make sure that your test store is eligible for creating shipping labels (has WCShip plugin installed and packages configured in WP Admin > WooCommerce > Settings > Shipping > Woocommerce Shipping).
2. Create an order that requires international shipping (origin country different from destination country).
3. Open app, navigate to Orders tab and select the created order.
4. Select Create Shipping Label.
5. Skip through Ship From, Ship To, Package Details to configure Customs.
6. Try enter random text in ITN field, notice that there's a validation error for the row saying "Invalid ITN format". If you enter a number with correct format (eg: AES X20080930987654) the error should go away.
7. Select Other for either Content Type or Restriction Type, notice that there's an error prompted asking you to enter more information about each. Entering non-empty text (not just white space) should make the error go away.

Update release notes:
